src/llm/mod.rs

```diff
@@ -48,6 +51,48 @@ pub trait LLM: Sync + Send {
     async fn generate(&self, prompt: Box<dyn Prompt>) -> Result<LLMResponse>;
 }
 
+/// Embedding trait is used to generate an embedding from an Online Service.
+#[async_trait::async_trait]
+pub trait EMBEDDING: Sync + Send {
+    /// Generate an embedding from an Online Service.
+    /// # Arguments
+    /// * `input` - A Record trait object.
+    ///
+    /// # Examples
+    /// This example uses the OpenAI chat models.
+    /// ```
+    /// use orca::llm::Embedding;
+    /// use orca::record::Record;
+    /// use orca::llm::openai::OpenAI;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///    let client = OpenAI::new();
+    ///    let input = Record::new("Hello, world");
+    ///    let response = client.generate_embedding(&input).await.unwrap();
+    ///    assert!(response.to_string().to_lowercase().contains("paris"));
+    /// }
+    /// ```
+    async fn generate_embedding<'a>(&'a self, input: &'a Record) -> Result<OpenAIEmbeddingResponse>;
```

Comment:
So, a `Record` is just a document (HTML, PDF, etc.) the user loads in the framework for an LLM to use, rather than a string. In this case, `generate_embeddings` should have a simple task, take in a prompt (text) and process it. For this reason, this function should take a `Box<dyn Prompt>` check out the prompt directory to see how they work. In essence, something that implements the `Prompt` trait will be able to call `to_vec` or `to_string` methods in order to be converted into the proper type. Sorry if this is confusing, I did this because certain models like GPT3.5 take in `Vec<Message>` while others such as `Bert` just take in a `String`, so dynamic dispatch seemed like a fitting solution at the time. Soon, I am planning to make changes so that `LLM::generate` (and `Embedding::generate_embedding`) methods just take in a `&str` to make it way simpler for the user, and just use templating to convert to `String` or `Vec<Message>` depending on the template. For now, changing `&'a Record` to `Box<dyn Prompt>` should be ok. Also using `Box` removes the need for lifetimes so the final definition could be as follows:
```rust
async fn generate_embedding(&self, input: Box<dyn Prompt>) -> Result<OpenAIEmbeddingResponse>;
```